### DISCUSSION TITLE
I can't install fooocus

### DISCUSSION THREAD

**Discussion by VanGuper** [LykosAI/StabilityMatrix#802] [Q&A]
When trying to install I get the following error:

Could not install Fooocus (StabilityMatrix.Core.Exceptions.ProcessException: pip install failed with code 1: 'Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu121\r\nCollecting torch==2.1.0\r\n  Using cached https://download.pytorch.org/whl/cu121/torch-2.1.0%2Bcu121-cp310-cp310-win_amd64.whl (2473.9 MB)\r\nCollecting torchvision==0.16.0\r\n  Using cached https://download.pytorch.org/whl/cu121/torchvision-0.16.0%2Bcu121-cp310-cp310-win_amd64.whl (5.8 MB)\r\nCollecting torchsde==0.2.6\r\n  Using cached torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)\r\nCollecting einops==0.8.0\r\n  Using cached einops-0.8.0-py3-none-any.whl.metadata (12 kB)\r\nCollecting transformers==4.42.4\r\n  Using cached transformers-4.42.4-py3-none-any.whl.metadata (43 kB)\r\nCollecting safetensors==0.4.3\r\n  Using cached safetensors-0.4.3-cp310-none-win_amd64.whl.metadata (3.9 kB)\r\nCollecting accelerate==0.32.1\r\n  Using cached accelerate-0.32.1-py3-none-any.whl.metadata (18 kB)\r\nCollecting pyyaml==6.0.1\r\n  Using cached PyYAML-6.0.1-cp310-cp310-win_amd64.whl.metadata (2.1 kB)\r\nCollecting pillow==10.4.0\r\n  Using cached pillow-10.4.0-cp310-cp310-win_amd64.whl.metadata (9.3 kB)\r\nCollecting scipy==1.14.0\r\n  Using cached scipy-1.14.0-cp310-cp310-win_amd64.whl.metadata (60 kB)\r\nCollecting tqdm==4.66.4\r\n  Using cached tqdm-4.66.4-py3-none-any.whl.metadata (57 kB)\r\nCollecting psutil==6.0.0\r\n  Using cached psutil-6.0.0-cp37-abi3-win_amd64.whl.metadata (22 kB)\r\nCollecting pytorch_lightning==2.3.3\r\n  Using cached pytorch_lightning-2.3.3-py3-none-any.whl.metadata (21 kB)\r\nCollecting omegaconf==2.3.0\r\n  Using cached omegaconf-2.3.0-py3-none-any.whl.metadata (3.9 kB)\r\nCollecting gradio==3.41.2\r\n  Using cached gradio-3.41.2-py3-none-any.whl.metadata (17 kB)\r\nCollecting pygit2==1.15.1\r\n  Using cached pygit2-1.15.1-cp310-cp310-win_amd64.whl.metadata (3.4 kB)\r\nCollecting opencv-contrib-python-headless==4.10.0.84\r\n  Using cached opencv_contrib_python_headless-4.10.0.84-cp37-abi3-win_amd64.whl.metadata (20 kB)\r\nCollecting httpx==0.27.0\r\n  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)\r\nCollecting onnxruntime==1.18.1\r\n  Using cached onnxruntime-1.18.1-cp310-cp310-win_amd64.whl.metadata (4.5 kB)\r\nCollecting timm==1.0.7\r\n  Using cached timm-1.0.7-py3-none-any.whl.metadata (47 kB)\r\nCollecting numpy==1.26.4\r\n  Using cached numpy-1.26.4-cp310-cp310-win_amd64.whl.metadata (61 kB)\r\nCollecting tokenizers==0.19.1\r\n  Using cached tokenizers-0.19.1-cp310-none-win_amd64.whl.metadata (6.9 kB)\r\nCollecting packaging==24.1\r\n  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)\r\nCollecting rembg==2.0.57\r\n  Using cached rembg-2.0.57-py3-none-any.whl.metadata (16 kB)\r\nCollecting groundingdino-py==0.4.0\r\n  Using cached groundingdino-py-0.4.0.tar.gz (82 kB)\r\n  Preparing metadata (setup.py): started\r\n  Preparing metadata (setup.py): finished with status 'done'\r\nCollecting segment_anything==1.0\r\n  Using cached segment_anything-1.0-py3-none-any.whl.metadata (487 bytes)\r\nCollecting filelock (from torch==2.1.0)\r\n  Using cached filelock-3.15.4-py3-none-any.whl.metadata (2.9 kB)\r\nCollecting typing-extensions (from torch==2.1.0)\r\n  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)\r\nCollecting sympy (from torch==2.1.0)\r\n  Using cached sympy-1.13.1-py3-none-any.whl.metadata (12 kB)\r\nCollecting networkx (from torch==2.1.0)\r\n  Using cached networkx-3.3-py3-none-any.whl.metadata (5.1 kB)\r\nCollecting jinja2 (from torch==2.1.0)\r\n  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)\r\nCollecting fsspec (from torch==2.1.0)\r\n  Using cached fsspec-2024.6.1-py3-none-any.whl.metadata (11 kB)\r\nCollecting requests (from torchvision==0.16.0)\r\n  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)\r\nCollecting trampoline>=0.1.2 (from torchsde==0.2.6)\r\n  Using cached trampoline-0.1.2-py3-none-any.whl.metadata (10 kB)\r\nCollecting huggingface-hub<1.0,>=0.23.2 (from transformers==4.42.4)\r\n  Using cached huggingface_hub-0.24.5-py3-none-any.whl.metadata (13 kB)\r\nCollecting regex!=2019.12.17 (from transformers==4.42.4)\r\n  Using cached regex-2024.7.24-cp310-cp310-win_amd64.whl.metadata (41 kB)\r\nCollecting colorama (from tqdm==4.66.4)\r\n  Using cached https://download.pytorch.org/whl/colorama-0.4.6-py2.py3-none-any.whl (25 kB)\r\nCollecting torchmetrics>=0.7.0 (from pytorch_lightning==2.3.3)\r\n  Using cached torchmetrics-1.4.1-py3-none-any.whl.metadata (20 kB)\r\nCollecting lightning-utilities>=0.10.0 (from pytorch_lightning==2.3.3)\r\n  Using cached lightning_utilities-0.11.6-py3-none-any.whl.metadata (5.2 kB)\r\nCollecting antlr4-python3-runtime==4.9.* (from omegaconf==2.3.0)\r\n  Using cached antlr4_python3_runtime-4.9.3-py3-none-any.whl\r\nCollecting aiofiles<24.0,>=22.0 (from gradio==3.41.2)\r\n  Using cached aiofiles-23.2.1-py3-none-any.whl.metadata (9.7 kB)\r\nCollecting altair<6.0,>=4.2.0 (from gradio==3.41.2)\r\n  Using cached altair-5.3.0-py3-none-any.whl.metadata (9.2 kB)\r\nCollecting fastapi (from gradio==3.41.2)\r\n  Using cached fastapi-0.112.0-py3-none-any.whl.metadata (27 kB)\r\nCollecting ffmpy (from gradio==3.41.2)\r\n  Using cached ffmpy-0.4.0-py3-none-any.whl.metadata (2.9 kB)\r\nCollecting gradio-client==0.5.0 (from gradio==3.41.2)\r\n  Using cached gradio_client-0.5.0-py3-none-any.whl.metadata (7.1 kB)\r\nCollecting importlib-resources<7.0,>=1.3 (from gradio==3.41.2)\r\n  Using cached importlib_resources-6.4.0-py3-none-any.whl.metadata (3.9 kB)\r\nCollecting markupsafe~=2.0 (from gradio==3.41.2)\r\n  Using cached https://download.pytorch.org/whl/MarkupSafe-2.1.5-cp310-cp310-win_amd64.whl (17 kB)\r\nCollecting matplotlib~=3.0 (from gradio==3.41.2)\r\n  Using cached matplotlib-3.9.1.tar.gz (36.1 MB)\r\n  Installing build dependencies: started\r\n  Installing build dependencies: finished with status 'done'\r\n  Getting requirements to build wheel: started\r\n  Getting requirements to build wheel: finished with status 'done'\r\n  Installing backend dependencies: started\r\n  Installing backend dependencies: finished with status 'done'\r\n  Preparing metadata (pyproject.toml): started\r\n  Preparing metadata (pyproject.toml): finished with status 'error'\r\n  error: subprocess-exited-with-error\r\n  \r\n  \u00d7 Preparing metadata (pyproject.toml) did not run successfully.\r\n  \u2502 exit code: 1\r\n  \u2570\u2500> [23 lines of output]\r\n      + meson setup C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e\.mesonpy-a0atgci2 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e\.mesonpy-a0atgci2\meson-python-native-file.ini\r\n      The Meson build system\r\n      Version: 1.5.1\r\n      Source dir: C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e\r\n      Build dir: C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e\.mesonpy-a0atgci2\r\n      Build type: native build\r\n      Program python3 found: YES\r\n      Project name: matplotlib\r\n      Project version: 3.9.1\r\n      Activating VS 16.11.38\r\n      C compiler for the host machine: cl (msvc 19.29.30154 "Compilador de optimizaci\u00a2n de C/C++ de Microsoft (R) versi\u00a2n 19.29.30154 para x64")\r\n      C linker for the host machine: link link 14.29.30154.0\r\n      C++ compiler for the host machine: cl (msvc 19.29.30154 "Compilador de optimizaci\u00a2n de C/C++ de Microsoft (R) versi\u00a2n 19.29.30154 para x64")\r\n      C++ linker for the host machine: link link 14.29.30154.0\r\n      Host machine cpu family: x86_64\r\n      Host machine cpu: x86_64\r\n      Program python found: YES (D:\SD IA\Data\Packages\Fooocus\venv\Scripts\python.exe)\r\n      Could not find Python3 library 'D:\\SD IA\\Data\\Packages\\Fooocus\\venv\\Scripts\\libs\\python310.lib'\r\n      Run-time dependency python found: NO (tried sysconfig)\r\n      \r\n      ..\meson.build:37:14: ERROR: Python dependency not found\r\n      \r\n      A full log can be found at C:\Users\VanGuper\AppData\Local\Temp\pip-install-t329okr9\matplotlib_181ac4cd599d45adb50221115484e33e\.mesonpy-a0atgci2\meson-logs\meson-log.txt\r\n      [end of output]\r\n  \r\n  note: This error originates from a subprocess, and is likely not a problem with pip.\r\nerror: metadata-generation-failed\r\n\r\n\u00d7 Encountered error while generating package metadata.\r\n\u2570\u2500> See above for output.\r\n\r\nnote: This is an issue with the package mentioned above, not pip.\r\nhint: See above for details.\r\n'
   at StabilityMatrix.Core.Python.PyVenvRunner.PipInstall(ProcessArgs args, Action`1 outputDataReceived)
   at StabilityMatrix.Core.Models.Packages.Fooocus.InstallPackage(String installLocation, TorchVersion torchVersion, SharedFolderMethod selectedSharedFolderMethod, DownloadPackageVersionOptions versionOptions, IProgress`1 progress, Action`1 onConsoleOutput)
   at StabilityMatrix.Core.Models.Packages.Fooocus.InstallPackage(String installLocation, TorchVersion torchVersion, SharedFolderMethod selectedSharedFolderMethod, DownloadPackageVersionOptions versionOptions, IProgress`1 progress, Action`1 onConsoleOutput)
   at StabilityMatrix.Core.Models.PackageModification.InstallPackageStep.ExecuteAsync(IProgress`1 progress)
   at StabilityMatrix.Core.Models.PackageModification.PackageModificationRunner.ExecuteSteps(IReadOnlyList`1 steps))

**Comment by mohnjiles** [comment]
Try the solution posted here: https://github.com/LykosAI/StabilityMatrix/issues/803#issuecomment-2268078788
